index.js

```diff
@@ -22,57 +19,85 @@ var hmac = sodium.crypto_auth
 
 exports.hash = u.hash
 
-exports.getTag = u.getTag
+exports.getFeedType = u.getFeedType
+exports.getTag = u.getFeedType // deprecated
 
 function isObject (o) {
   return 'object' === typeof o
 }
 
-function isFunction (f) {
-  return 'function' === typeof f
-}
-
 function isString(s) {
   return 'string' === typeof s
 }
 
-var curves = {}
-curves.ed25519 = require('./sodium')
+const feedTypes = {
+  ed25519: require('./sodium'),
+  'ed25519.test': require('./sodium')
+}
+
+exports.use = (name, object) => {
+  if (typeof name !== 'string' || name.length === 0) {
+    throw new Error(`Invalid name: "${name}", expected string with non-zero length`)
+  }
+
+  const requiredMethods = [
+    'generate',
+    'sign',
+    'verify'
+  ]
+
+  const isNotObject = typeof object !== 'object'
+  const isInvalidObject = isNotObject || requiredMethods.some(methodName => 
+    typeof object[methodName] !== 'function'
+  )
+
+  if (isInvalidObject) {
+    const expectedMethods = requiredMethods.join(', ')
+    console.log(object)
+    throw new Error(`Invalid object. Missing required methods, expected: ${expectedMethods}`)
+  }
+
+  if (feedTypes[name] != null) {
+    throw new Error(`Duplicate feed type: "${name}"`)
+  }
+
+  feedTypes[name] = object
+}
 
-function getCurve(keys) {
-  var curve = keys.curve
+function getFeedType(keys) {
+  let { feedType } = keys
+  feedType = feedType || keys.curve
 
-  if(!keys.curve && isString(keys.public))
+  if(!feedType && isString(keys.public))
     keys = keys.public
 
-  if(!curve && isString(keys))
-    curve = u.getTag(keys)
+  if(!feedType && isString(keys))
+    feedType = u.getFeedType(keys)
 
-  if(!curves[curve]) {
-    throw new Error(
-      'unkown curve:' + curve +
-      ' expected: '+Object.keys(curves)
-    )
+  if(!feedTypes[feedType]) {
+    throw new Error(`unkown feed type: "${feedType}", expected: "${Object.keys(feedTypes)}"`)
   }
 
-  return curve
+  return feedType
 }
 
 //this should return a key pair:
-// {curve: curve, public: Buffer, private: Buffer}
+// { feedType: string, curve: string, public: Buffer, private: Buffer}
```

Comment:
If this is a known requirement, then why don't you check this is what the `generate` does when you register a new type with the `use()` method?